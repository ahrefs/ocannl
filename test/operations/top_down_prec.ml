(* A simple demo of using the Threefry4x32 PRNG in OCANNL *)

open Base
module Tensor = Ocannl.Tensor
module Train = Ocannl.Train
module TDSL = Ocannl.Operation.TDSL
module Tn = Ir.Tnode

let () =
  Tensor.unsafe_reinitialize ();
  let module Backend = (val Backends.fresh_backend ()) in
  let%op d = ("a" [2] + "b" [2]) *. "c" [2] in
  Tn.update_prec b.value Ir.Ops.half;
  Tn.update_prec d.value Ir.Ops.bfloat16;
  (* Compile and run *)
  Ocannl.Train.set_hosted d.value;
  ignore (Ocannl.Train.forward_once (module Backend) d);
  Train.printf d
