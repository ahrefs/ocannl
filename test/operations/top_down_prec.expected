Retrieving commandline, environment, or config file variable ocannl_log_level
Found 0, in the config file
┌────────────────────┐
│[8]: *._d shape 0:1 │
│┌┬──────┐           │
│││axis 0│           │
│├┼──────┤           │
│││ 8.00 │           │
│└┴──────┘           │
└────────────────────┘
grad_*._d <not-hosted>
